### PR TITLE
Improve PyAPI loading of cross sections from HDF5

### PR DESCRIPTION
**Pull request by smharper**
This PR makes some minor changes to the recently merged HDF5 data features.
* Fix bug where reaction thresholds were not used
* Automatically build summed reactions
* Allow Tab1 evaluations of un-ordered energies

### CODE REVIEW

File: openmc/data/neutron.py [hunk 1]
```diff
@@ -326,6 +326,17 @@ def from_hdf5(cls, group_or_filename):
                     tgroup = group['total_nu']
                     rx.derived_products.append(Product.from_hdf5(tgroup))
 
+        # Build summed reactions.  Start from the highest MT number because high
+        # MTs never depend on lower MTs.
+        for mt_sum in sorted(SUM_RULES.keys())[::-1]:
```
Comment (paulromano) [inline openmc/data/neutron.py]:
```python
sorted(..., reverse=True)
```

Comment (paulromano) [inline openmc/data/neutron.py]: And actually, sorting on a dictionary uses key by default so you should be able to just do
```Python
sorted(SUM_RULES, reverse=True)
```

File: openmc/data/reaction.py [hunk 2]
```diff
@@ -400,7 +400,7 @@ def from_hdf5(cls, group, energy):
         # Read cross section
         if 'xs' in group:
             xs = group['xs'].value
-            rx.xs = Tabulated1D(energy, xs)
+            rx.xs = Tabulated1D(energy[rx.threshold_idx:], xs)
```
Comment (paulromano) [inline openmc/data/reaction.py]: Ooo, nice catch

**Comment by smharper** [comment]
As discussed in person, I modified the notebook showing how to call the `xs` functions which works even for summed reactions.  I had to do some re-ordering to make it fit well, and then I added a little more content.  Have a look and let me know if should be focused more tightly on the `from_ace` side of the data module.

**Comment by paulromano** [comment]
Thanks @smharper! I like the changes you made in the notebook.

**Pull request merged**